### PR TITLE
fix: autoCancel respected when hook is unmounted

### LINKED ISSUE #609: autoCancel:false not working
In some specific requests we want to prevent autoCancel when navigating to different route thought the react-router-dom. 

sample code

`const [{ data, loading, error, response }, requestHandler] = useAPI(
    {method: "POST",
      url: "/selections",
      data: {
        id: id,
      }},
    { manual: true, autoCancel:false }
  );`

another try with configuration
`
const defaultOptions  = { manual: false, useCache: true, ssr: true, autoCancel: false}
configure({ axios, cache, defaultOptions});
`
But both methods didn't work. 

Even directly changing node module like below
`
const DEFAULT_OPTIONS = {
  manual: false,
  useCache: true,
  ssr: true,
  autoCancel: false
}
`

Appreciate if you can look into this.

### PR DESCRIPTION
**Pull request by lukaskral**
this code change fixes issue described in #609 

fixes #609

### CODE REVIEW

**Comment by simoneb** [comment]
Can you add a test to cover this scenario?

**Comment by lukaskral** [comment]
unit test added

File: test/index.test.jsx [hunk 1]
```diff
@@ -387,6 +387,20 @@ function standardTests(
         expect(cancel).not.toHaveBeenCalled()
       })
 
+      it.only('should skip default cancellation after unmount if options.autoCancel is set to false', async () => {
```
Comment (simoneb) [inline test/index.test.jsx:390]: remove .only

**Review by simoneb — CHANGES_REQUESTED** [review]

**Review by simoneb — APPROVED** [review]

**Comment by lukaskral** [comment]
@simoneb thanks for super fast responses 👏

**Comment by simoneb** [comment]
No problem, thank you for the contribution!

**Pull request merged**